pkg/httputils/http_util.go

```diff
@@ -210,16 +213,37 @@ func Do(url string, headers map[string]string, timeout time.Duration) (string, e
 
 // HTTPGet sends an HTTP GET request with headers.
 func HTTPGet(url string, headers map[string]string) (*http.Response, error) {
-	return HTTPWithHeaders("GET", url, headers, 0)
+	return HTTPWithHeaders("GET", url, headers, 0, nil)
 }
 
 // HTTPGetTimeout sends an HTTP GET request with timeout.
 func HTTPGetTimeout(url string, headers map[string]string, timeout time.Duration) (*http.Response, error) {
-	return HTTPWithHeaders("GET", url, headers, timeout)
+	return HTTPWithHeaders("GET", url, headers, timeout, nil)
+}
+
+// HTTPGetWithTLS sends an HTTP GET request with TLS config.
+func HTTPGetWithTLS(url string, headers map[string]string, timeout time.Duration, cacerts []string, insecure bool) (*http.Response, error) {
+	roots := x509.NewCertPool()
+	appendSuccess := false
+	for _, certPath := range cacerts {
+		certBytes, err := ioutil.ReadFile(certPath)
+		if err != nil {
+			return nil, err
+		}
+		appendSuccess = appendSuccess || roots.AppendCertsFromPEM(certBytes)
+	}
+
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: insecure,
+	}
+	if appendSuccess {
+		tlsConfig.RootCAs = roots
+	}
+	return HTTPWithHeaders("GET", url, headers, timeout, tlsConfig)
 }
 
 // HTTPWithHeaders sends an HTTP request with headers and specified method.
-func HTTPWithHeaders(method, url string, headers map[string]string, timeout time.Duration) (*http.Response, error) {
+func HTTPWithHeaders(method, url string, headers map[string]string, timeout time.Duration, tlsConfig *tls.Config) (*http.Response, error) {
```

Comment:
> I think you can add a new function to wrap the logic here. And then you will no need to update the functions such as `HTTPGet`, etc.

yes, I can add a new func that wrapper `HTTPWithHeaders`, but I can not avoid to change `HTTPWithHeaders`. Because tls config needs to be passed into http client.